skymp5-server/cpp/server_guest_lib/MpObjectReference.cpp

```diff
@@ -908,6 +908,56 @@ void MpObjectReference::RegisterPrivateIndexedProperty(
   }
 }
 
+void MpObjectReference::AddToFaction(Faction faction)
```

Comment:
In Skyrim ObjectReferences can't have faction ranks. Only Actors can. 

By the way AddToFaction is a good place to lazily read factions from NPC_ record